### PR TITLE
weather-* scripts default to runtime container image.

### PR DESCRIPTION
**Pull request by DarshanSP19**
- Removed `--extra_package` and relevant build logic from all three tools.
- Injecting the `args` related to the runtime container (`--sdk_container_image` and `--experiments`) if not provided by the user.
- `--sdk_container_image` will be the public docker image like below.
- New flag for suppling the local code to DataflowRunner `-–use-local-code`.
```
gcr.io/weather-tools-prod/weather-tools:0.0.0
```

- Resolves the first part of #353

### CODE REVIEW

File: weather_dl/download_pipeline/pipeline.py [hunk 1]
```diff
@@ -46,6 +46,7 @@
 from .stores import TempFileStore, LocalFileStore
 
 logger = logging.getLogger(__name__)
+SDK_CONTAINER_IMAGE='gcr.io/weather-tools-prod/weather-tools:0.0.0'
```
Comment (alxmrs) [inline weather_dl/download_pipeline/pipeline.py:49]: Let's make this an environment variable with this GCR image as the default.

File: weather_dl/download_pipeline/pipeline.py [hunk 2]
```diff
@@ -180,6 +181,10 @@ def run(argv: t.List[str], save_main_session: bool = True) -> PipelineArgs:
 
     known_args, pipeline_args = parser.parse_known_args(argv[1:])
 
+    if "DataflowRunner" in pipeline_args and  "--sdk_container_image" not in pipeline_args:
```
Comment (alxmrs) [inline weather_dl/download_pipeline/pipeline.py:184]: I think this should go in the `weather-dl` script.

File: weather_dl/download_pipeline/pipeline.py (hunk 1 above)
Comment (alxmrs) [reply weather_dl/download_pipeline/pipeline.py:49]: Ah, I see that you're doing this below. Great!

File: weather_dl/weather-dl [hunk 3]
```diff
@@ -49,28 +45,4 @@ if __name__ == '__main__':
...
-
-            with tarfile.open(pkg_archive, 'r') as tar:
-                assert any([f.endswith('.py') for f in tar.getnames()]), 'extra_package must include python files!'
-
-            # cleanup memory to prevent pickling error.
-            tar = None
-            weather_dl = None
-
-            cli(['--extra_package', pkg_archive])
+    cli()
```
Comment (alxmrs) [inline weather_dl/weather-dl:48]: As mentioned above, I think the SDK container argument should be input here as a few list arguments in the CLI.

**Review by alxmrs — COMMENTED** [review]
Left comments for how weather-dl should be structured; the same applies to the other tools. 

Thanks so much for this! Really excited to see this feature land.

File: weather_dl/download_pipeline/pipeline.py (hunk 2 above)
Comment (DarshanSP19) [reply weather_dl/download_pipeline/pipeline.py]: Updated.

File: weather_dl/weather-dl (hunk 3 above)
Comment (DarshanSP19) [reply weather_dl/weather-dl]: Updated.

File: weather_dl/download_pipeline/pipeline.py [hunk 4]
```diff
@@ -177,6 +177,7 @@ def run(argv: t.List[str], save_main_session: bool = True) -> PipelineArgs:
                         help="Update the manifest for the already downloaded shards and exit. Default: 'false'.")
     parser.add_argument('--log-level', type=int, default=2,
                       help='An integer to configure log level. Default: 2(INFO)')
+    parser.add_argument('--use-local-code', action='store_true', default=False, help='Supply local code to the Runner.')
```
Comment (alxmrs) [inline weather_dl/download_pipeline/pipeline.py:180]: Please update the README docs with this flag. The way I do this is by running copy & pasting from the help output (`--help`).

**Review by alxmrs — APPROVED** [review]
One nit, but then this is good to go!

File: weather_dl/download_pipeline/pipeline.py (hunk 4 above)
Comment (DarshanSP19) [reply weather_dl/download_pipeline/pipeline.py:180]: Updated.

**Pull request merged**